tests/ui_tests/test_ui_add_component_window.py

```diff
@@ -1000,3 +1649,84 @@ def test_UI_GIVEN_field_widget_with_string_type_THEN_value_property_is_correct(q
 
     assert field.name == field_name
     assert field.value[...] == field_value
+
+
+def enter_file_path(
+    qtbot: pytestqt.qtbot.QtBot,
+    dialog: AddComponentDialog,
+    template: PySide2.QtWidgets.QDialog,
+    file_path: str,
+    file_contents: str,
+):
+    """
+    Mimics the user entering a file path. Mimics a button click and patches the methods that deal with loading a
+    geometry file.
+    :param qtbot: The qtbot testing tool.
+    :param dialog: An instance of an AddComponentDialog.
+    :param template: The window/widget that holds the AddComponentDialog.
+    :param file_path: The desired file path.
+    :param file_contents: The file contents that are returned by the open mock.
+    """
+    with patch(
+        "nexus_constructor.add_component_window.file_dialog", return_value=file_path
+    ):
+        with patch("builtins.open", mock_open(read_data=file_contents)):
+            systematic_button_press(qtbot, template, dialog.fileBrowseButton)
+
+
+def enter_component_name(
+    qtbot: pytestqt.qtbot.QtBot,
+    template: PySide2.QtWidgets.QDialog,
+    dialog: AddComponentDialog,
+    component_name: str,
+):
+    """
+    Mimics the user entering a component name in the Add Component dialog. Clicks on the text field and enters a given
+    name.
+    :param qtbot: The qtbot testing tool.
+    :param dialog: An instance of an AddComponentDialog.
+    :param template: The window/widget that holds the AddComponentDialog.
+    :param component_name: The desired component name.
+    """
+    qtbot.mouseClick(dialog.nameLineEdit, Qt.LeftButton)
+    qtbot.keyClicks(dialog.nameLineEdit, component_name)
+    show_and_close_window(qtbot, template)
+
+
+def systematic_button_press(
```

Comment:
These are generic helper functions for ui tests, they should be in a separate file.